include/umf/providers/provider_fixed_memory.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#ifndef UMF_FIXED_MEMORY_PROVIDER_H
+#define UMF_FIXED_MEMORY_PROVIDER_H
+
+#include <umf/providers/provider_os_memory.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// @cond
+#define UMF_FIXED_RESULTS_START_FROM 4000
+/// @endcond
+
+struct umf_fixed_memory_provider_params_t;
+
+typedef struct umf_fixed_memory_provider_params_t
+    *umf_fixed_memory_provider_params_handle_t;
+
+/// @brief  Create a struct to store parameters of the Fixed Memory Provider.
```

Comment:
pls add new provider to `api.rst` web docs

Reply:
also:
1. remove Coarse Provider info
2. add not only web docs, but also info in README